### DISCUSSION TITLE
3D foam modelling visual representation

### DISCUSSION THREAD

**Discussion by ap264** [neperfepx/neper#550] [Q&A] [answered]
Dear Sir

I would like to know if it is possible for me to represent my 3d mesh model like this 

![Capture1](https://user-images.githubusercontent.com/109927295/187251842-cb072521-e34d-4b3a-b34e-214b7c807f4d.PNG)

As I import my model into Abaqus, the foam inside the cube disappears and it looks something like this (transparent view of the cube)

![Capture](https://user-images.githubusercontent.com/109927295/187252273-14ec5a1a-1c75-4efd-8f9c-f765c733ad95.PNG)

Commands used to generate this model are as follows

neper -T n 79 -morpho "diameq:custom(mydistibu)" -o mytess
neper -M mytess.tess -elttype tet -rcl 0.02 -format inp

I acknowledge that I could achieve this type of result with 2d meshing but i would like to know if it possible with 3d meshing.

**Comment by rquey** [comment, marked as answer]
Use `-showelt2d all` or (`-showelt2d domtype!=2`) and `-showelt3d none`.

**Reply by ap264** [reply]
I have tried using showelt2d option but i get the error in my module saying that "showelt2d is bad argument". 

attached is the text of the above mentioned error 

```
akshay@akshay-Inspiron-15-3567:~/Desktop/neper/voronoi_research$ neper -M n79-id1.tess -elttype tet -rcl 0.02 -showelt2d all -showelt3d none -format inp

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.4.2-33
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 4 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/akshay/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n79-id1.tess -elttype tet -rcl 0.02 -showelt2d all
         -showelt3d none -format inp
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Error  : Unknown option `-showelt2d'.
Error  : Bad arguments!

Aborted (core dumped)
```
